app/src/examples/RuntimeTests/tests/core/useDerivedValue/basic.test.tsx

```diff
@@ -0,0 +1,232 @@
+} from '../../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+import { BasicSnapshots } from './useDerivedValue.snapshot';
+
+const WIDTH_COMPONENT = 'WidthComponent';
+
+describe('Test useDerivedValue changing width', () => {
+  enum AnimationLocation {
+    NONE = 'no animation',
+    USE_EFFECT = 'animation inside useEffect',
+    ANIMATED_STYLE = 'animation inside useAnimatedStyle',
+  }
+
+  enum AnimationType {
+    NONE = '',
+    TIMING = 'timing',
+    SPRING = 'spring',
+  }
+
+  const WidthComponent = ({
+    startWidth,
+    finalWidth,
+    animate,
+    animationType,
+    deriveFunction,
+  }: {
+    startWidth: number;
+    finalWidth: number;
+    animate: AnimationLocation;
+    animationType: AnimationType;
+    deriveFunction: (a: number) => number;
+  }) => {
+    const basicValue = useSharedValue(startWidth);
+    const componentRef = useTestRef(WIDTH_COMPONENT);
+    const derivedValue = useDerivedValue(() => {
+      return deriveFunction(basicValue.value);
+    });
+
+    const derivedValueStyle = useAnimatedStyle(() => {
+      if (animate === AnimationLocation.ANIMATED_STYLE) {
+        return {
+          width:
+            animationType === AnimationType.TIMING ? withTiming(derivedValue.value) : withSpring(derivedValue.value),
+        };
+      } else {
+        return { width: derivedValue.value };
+      }
+    });
+
+    useEffect(() => {
+      if (animate === AnimationLocation.USE_EFFECT) {
+        basicValue.value = animationType === AnimationType.TIMING ? withTiming(finalWidth) : withSpring(finalWidth);
+      } else {
+        basicValue.value = finalWidth;
+      }
+    }, [basicValue, finalWidth, animate, animationType]);
+
+    return (
+      <View style={styles.container}>
+        <Animated.View ref={componentRef} style={[styles.animatedBox, derivedValueStyle]} />
+      </View>
+    );
+  };
+
+  [
+    {
+      derivedFun: (x: number) => {
+        'worklet';
+        return 2 * x + 20;
+      },
+      msg: 'x => 2 * x + 20',
+      index: 0,
+    },
+
+    {
+      derivedFun: (x: number) => {
+        'worklet';
+        return 100 + Math.sin(x) * 80;
+      },
+      msg: 'x => 100 + 80sin(x)',
+      index: 1,
+    },
+  ].forEach(({ derivedFun, msg, index }) => {
+    describe(`Tests for derived function ${msg}`, () => {
+      async function getSnapshotUpdatesAndCheckFinalValue(
+        startWidth: number,
+        finalWidth: number,
+        animate: AnimationLocation,
+        animationType: AnimationType,
+      ) {
+        await mockAnimationTimer();
+        const updatesContainerActive = await recordAnimationUpdates();
+        await render(
+          <WidthComponent
+            startWidth={startWidth}
+            finalWidth={finalWidth}
+            animate={animate}
+            animationType={animationType}
+            deriveFunction={derivedFun}
+          />,
+        );
+        const testComponent = getTestComponent(WIDTH_COMPONENT);
+        await wait(1800);
```

Comment:
can we make it a bit faster?

Reply:
The delay may depend on test case, so we won't have unnecessary dalay for quick cases.